integration/src/test/java/edu/unc/lib/boxc/integration/web/access/SearchEndpointIT.java

```diff
@@ -433,6 +434,112 @@ public void testSearchWithSubjectFilterSelectedDisplaysAvailableLanguages() thro
         }
     }
 
+    @Test
+    public void testGetFacetsFalseSearchReturnsNoFacets() throws Exception {
+        createDefaultObjects();
+        createLanguageSubjectObjects();
+
+        var getMethod = new HttpGet(SEARCH_URL + "/?getFacets=false");
+
+        try (var resp = httpClient.execute(getMethod)) {
+            ObjectMapper mapper = new ObjectMapper();
+            var respJson = mapper.readTree(resp.getEntity().getContent());
+
+            assertSuccessfulResponse(resp);
+            assertFalse(respJson.has("facetFields"));
+        }
+    }
+
+    @Test
+    public void testGetFacetsTrueSearchReturnsDefaultFacets() throws Exception {
+        createDefaultObjects();
+        var getMethod = new HttpGet(SEARCH_URL + "/?getFacets=true");
+
+        try (var resp = httpClient.execute(getMethod)) {
+            var facetFields = getFacetsFromResponse(resp);
+            var facetNames = new ArrayList<String>();
+
+            for (JsonNode facet : facetFields) {
+                facetNames.add(facet.get("name").asText());
+            }
+
+            assertSuccessfulResponse(resp);
+            assertEquals(10,facetFields.size());
+            assertEquals(defaultFacets(), facetNames);
+        }
+    }
+
+    @Test
+    public void testSearchWithFacetSelectOnlyShowsThoseFacets() throws Exception {
+        createDefaultObjects();
+        var getMethod = new HttpGet(SEARCH_URL + "/?facetSelect=language%2Csubject&getFacets=true");
+
+        try (var resp = httpClient.execute(getMethod)) {
+            var facetFields = getFacetsFromResponse(resp);
+
+            assertSuccessfulResponse(resp);
+            assertEquals(2,facetFields.size());
+            assertEquals("LANGUAGE", facetFields.get(0).get("name").asText());
+            assertEquals("SUBJECT", facetFields.get(1).get("name").asText());
+        }
+    }
+
+    @Test
+    public void testSearchWithFacetSelectCannotShowDisallowedFacets() throws Exception {
+        createDefaultObjects();
+        var getMethod = new HttpGet(SEARCH_URL + "/?facetSelect=language,subject,content+status&getFacets=true");
```

Comment:
the name for the CONTENT_STATUS parameter is `contentStatus`. The URL forms of the field names are listed in SearchFieldKey in the `urlParam` field.